### PR TITLE
Restore iter range pruning on comprehension ASTs

### LINKED ISSUE #735: interpreter.PruneAst does not prune `IterRange` expression
**Describe the bug**
interpreter.PruneAst function ignores the `IterRange` field of a comprehension expression.

**To Reproduce**
Check which components this affects:

- [ ]  parser
- [ ]  checker
- [x]  interpreter

Sample expression and input that reproduces the issue:

`users` is a list of const strings, and `R` is an unknown attribute pattern.

```cel
users.filter(u, u.startsWith(R.attr.prefix))
```

Test setup:
```go
package main

import (
	"fmt"
	"log"

	"github.com/google/cel-go/cel"
	"github.com/google/cel-go/checker/decls"
	"github.com/google/cel-go/interpreter"
	expr "google.golang.org/genproto/googleapis/api/expr/v1alpha1"
)

func main() {
	env, err := cel.NewEnv(cel.Declarations(
		decls.NewVar("R", decls.NewMapType(decls.String, decls.Dyn)),
		decls.NewVar("users", decls.NewListType(decls.String)),
	))
	exitOnErr(err)

	ast, iss := env.Compile(`users.filter(u, u.startsWith(R.attr.prefix))`)
	exitOnErr(iss.Err())

	prg, err := env.Program(ast, cel.EvalOptions(cel.OptTrackState, cel.OptPartialEval))
	exitOnErr(err)

	in := map[string]any{
		"users": []string{"alice", "bob"},
	}
	vars, err := cel.PartialVars(in, cel.AttributePattern("R").QualString("attr").Wildcard())
	exitOnErr(err)

	_, details, err := prg.Eval(vars)
	exitOnErr(err)

	pruned := interpreter.PruneAst(ast.Expr(), ast.SourceInfo().GetMacroCalls(), details.State()).Expr
	ce := pruned.ExprKind.(*expr.Expr_ComprehensionExpr).ComprehensionExpr
	printExpr("IterRange:", ce.IterRange)
}

func printExpr(name string, e *expr.Expr) {
	prunedAST := cel.ParsedExprToAst(&expr.ParsedExpr{Expr: e})
	s, err := cel.AstToString(prunedAST)
	exitOnErr(err)
	fmt.Println(name, s)
}

func exitOnErr(err error) {
	if err != nil {
		log.Fatalf("Error: %v", err)
	}
}

```

**Expected behavior**
The output should be:

```
IterRange: ["alice", "bob"]
```

instead of
```
IterRange: users
```

**Additional context**
This works as expected in v0.15.2, but in v0.15.3, the pruner [was modified](https://github.com/google/cel-go/pull/705/files#diff-531f45e46499da01fa105d485bd3d5765123ee5624c4128cad21545ce18bce25L411-L431) to skip the `IterRange` expression. Is this a breaking change?

### PR DESCRIPTION
**Pull request by TristonianJones**
Closes #735 and adds a small test to ensure the AST transformation is preserved.

### CODE REVIEW

File: interpreter/prune_test.go [hunk 1]
```diff
@@ -466,9 +468,22 @@ func TestPrune(t *testing.T) {
 			ExhaustiveEval(), Observe(EvalStateObserver(state)))
 		interpretable.Eval(testActivation(t, tst.in))
 		newExpr := PruneAst(ast.GetExpr(), ast.GetSourceInfo().GetMacroCalls(), state)
+		if tst.iterRange != "" {
+			compre := newExpr.GetExpr().GetComprehensionExpr()
+			if compre == nil {
+				t.Fatal("iter range check cannot operator on non comprehension output")
```
Comment (l46kok) [inline interpreter/prune_test.go]: nit:
```suggestion
				t.Fatal("iter range check cannot operate on non comprehension output")
```

File: interpreter/prune_test.go [hunk 2]
```diff
@@ -466,9 +468,22 @@ func TestPrune(t *testing.T) {
 			ExhaustiveEval(), Observe(EvalStateObserver(state)))
 		interpretable.Eval(testActivation(t, tst.in))
 		newExpr := PruneAst(ast.GetExpr(), ast.GetSourceInfo().GetMacroCalls(), state)
+		if tst.iterRange != "" {
```
Comment (l46kok) [inline interpreter/prune_test.go:474]: Up to you: since this check only applies for a single test, consider pulling this out into a separate test.

**Review by l46kok — DISMISSED** [review]

File: interpreter/prune_test.go (hunk 2 above)
Comment (TristonianJones) [reply interpreter/prune_test.go:474]: I added some additional tests. PTAL

File: interpreter/prune_test.go (hunk 1 above)
Comment (TristonianJones) [reply interpreter/prune_test.go]: Done.

**Comment by TristonianJones** [comment]
@l46kok PTAL. Thanks!

**Review by l46kok — APPROVED** [review]

**Pull request merged**